terraform-linter panic on end-of-line comment

### TL;DR

When a tf file contains an end-of-line comment, terraform-linter panics.

I've pushed a branch `tflint_panic_repro` containing a test case that reproduces the issue:

https://github.com/abcxyz/pkg/blob/80d4df595bbc5e820de7776616e8949b2696707e/internal/tools/terraformlinter/terraform_linter_test.go#L467-L475

The stack trace is:

```
panic: runtime error: slice bounds out of range [1:0] [recovered]
	panic: runtime error: slice bounds out of range [1:0]

goroutine 42 [running]:
testing.tRunner.func1.2({0x6dc440, 0xc0005a1bf0})
	/usr/lib/google-golang/src/testing/testing.go:1533 +0x24e
testing.tRunner.func1()
	/usr/lib/google-golang/src/testing/testing.go:1536 +0x39f
panic({0x6dc440, 0xc0005a1bf0})
	/usr/lib/google-golang/src/runtime/panic.go:884 +0x213
github.com/abcxyz/pkg/internal/tools/terraformlinter.validateBlock({0xc00009ec60?, 0xc000382340?, 0x8?})
	/usr/local/google/home/revell/git/pkg/internal/tools/terraformlinter/terraform_linter.go:219 +0x57f
github.com/abcxyz/pkg/internal/tools/terraformlinter.findViolations({0xc00001e200?, 0x701426?, 0x31?}, {0x0?, 0x77b169?})
	/usr/local/google/home/revell/git/pkg/internal/tools/terraformlinter/terraform_linter.go:192 +0x24b
github.com/abcxyz/pkg/internal/tools/terraformlinter.TestTerraformLinter_FindViolations.func1(0xc0001e1a00)
	/usr/local/google/home/revell/git/pkg/internal/tools/terraformlinter/terraform_linter_test.go:576 +0x65
testing.tRunner(0xc0001e1a00, 0xc000129320)
	/usr/lib/google-golang/src/testing/testing.go:1583 +0x10b
created by testing.(*T).Run
	/usr/lib/google-golang/src/testing/testing.go:1636 +0x3ea
exit status 2
FAIL	github.com/abcxyz/pkg/internal/tools/terraformlinter	0.036s
```

### Expected behavior

_No response_

### Observed behavior

_No response_

### Environment Details

```markdown
N/A
```


### Additional information

_No response_

--- a/internal/tools/terraformlinter/terraform_linter.go
+++ b/internal/tools/terraformlinter/terraform_linter.go
@@ -214,7 +214,8 @@ func validateBlock(tokens hclsyntax.Tokens) []*ViolationInstance {
 			var t hclsyntax.Token
 			skipping := true
 			depth := 0
-			for skipping {
+			// while there are tokens to skip and we haven't exceeded the length of the slice
+			for skipping && len(tokens) > 1 {
 				t, tokens = tokens[0], tokens[1:]
 				if t.Type == hclsyntax.TokenOBrace || t.Type == hclsyntax.TokenOBrack {
 					depth = depth + 1
@@ -222,7 +223,7 @@ func validateBlock(tokens hclsyntax.Tokens) []*ViolationInstance {
 				if t.Type == hclsyntax.TokenCBrace || t.Type == hclsyntax.TokenCBrack {
 					depth = depth - 1
 				}
-				if depth == 0 && t.Type == hclsyntax.TokenNewline {
+				if depth == 0 && (t.Type == hclsyntax.TokenNewline || t.Type == hclsyntax.TokenComment) {
 					// Check for an extra newline
 					trailingNewline := false
 					if len(tokens) > 0 && tokens[0].Type == hclsyntax.TokenNewline {

--- a/internal/tools/terraformlinter/terraform_linter_test.go
+++ b/internal/tools/terraformlinter/terraform_linter_test.go
@@ -556,6 +556,19 @@ func TestTerraformLinter_FindViolations(t *testing.T) {
 			},
 			wantError: false,
 		},
+		// Terraform AST treats comments on a line differently than any other token.
+		// Comments absorb the newline character instead of treating it as a separate token.
+		// This requires us to check for either a true newline token or a comment token
+		// that we can treat as the end of the line.
+		{
+			name: "repro_panic_on_comment_at_end_of_line",
+			content: `
+				resource "a" "b" {
+					c = var.d # e
+				}
+			`,
+			wantError: false,
+		},
 	}
 
 	for _, tc := range cases {

